res/controllers/Reloop-Digital-Jockey-2-IE.scripts.js

```diff
@@ -0,0 +1,876 @@
+        effectUnit: 1,
+        group: "[Channel" + number + "]",
+    });
+    this.fx2AssignmentButton = new components.EffectAssignmentButton({
+        midi: [0x90, RDJ2.BUTTONMAP_CH0_CH1.fx2assign[number - 1]],
+        effectUnit: 2,
+        group: "[Channel" + number + "]",
+    });
+
+    // high kill / quick effect enable button
+    this.highKillQuickEffectButton = new RDJ2.HighKillQuickEffectButton({
+        midi: [0x90, RDJ2.BUTTONMAP_CH0_CH1.highkill[number - 1]],
+        channelNr: number,
+    });
+
+
+    // Set the group properties of the above Components and connect their output callback functions
+    // Without this, the group property for each Component would have to be specified to its
+    // constructor.
+    this.reconnectComponents(function(component) {
+        if (component.group === undefined) {
+            // 'this' inside a function passed to reconnectComponents refers to the ComponentContainer
+            // so 'this' refers to the custom Deck object being constructed
+            component.group = this.currentDeck;
+        }
+    });
+};
+
+// give our custom Deck all the methods of the generic Deck in the Components library
+RDJ2.Deck.prototype = Object.create(components.Deck.prototype);
+
+/* get/set Values */
+
+RDJ2.Deck.prototype.getValue = function(key) {
+    return engine.getValue(this.group, key);
+};
+
+RDJ2.Deck.prototype.setValue = function(key, value) {
+    engine.setValue(this.group, key, value);
+};
+
+/* Load Track */
+
+RDJ2.LoadButton = function(options) {
+    components.Button.call(this, options);
+};
+RDJ2.LoadButton.prototype = new components.Button({
+    outKey: "track_loaded",
+    unshift: function() {
+        this.inKey = "LoadSelectedTrack";
+    },
+    shift: function() {
+        this.inKey = "eject";
+    },
+});
+
+/* Cue & Play */
+
+RDJ2.Deck.prototype.isPlaying = function() {
+    return this.getValue("play");
+};
+
+/* Pitch Bend / Track Search */
+
+RDJ2.BendMinusButton = function(options) {
+    components.Button.call(this, options);
+};
+RDJ2.BendMinusButton.prototype = new components.Button({
+    input: function(channel, control, value, status) {
+        var isPlaying = engine.getValue(this.group, "play");
+        if (isPlaying) {
+            engine.setValue(this.group, "back", false);
+            this.inSetValue(this.isPress(channel, control, value, status));
+        } else {
+            engine.setValue(this.group, "back", this.isPress(channel, control, value, status));
+        }
+    },
+    unshift: function() {
+        this.inSetValue(false);     // bug? inKey was not reset by default on unshift()
+        this.inKey = "rate_temp_down";
+        this.disconnect();
+        this.outKey = "rate_temp_down";
+        this.connect();
+        this.trigger();
+    },
+    /* The shifted action is disabled here as it has been remapped
+       to '[Library]GoToItem' normal binding in the xml file.
```

Comment:
can you confirm this behavior does not conflict? Also if you only have an `unshift`, you could just defined `this.key = rate_temp_down` in the constructor IIUC.

Reply:
I have simplified it and applied your suggestions in the last commit.